hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/placement/algorithms/TestSCMContainerPlacementRackAware.java

```diff
@@ -588,4 +588,213 @@ public void testOutOfServiceNodesNotSelected(int datanodeCount) {
       dnInfos.get(index).setNodeStatus(new NodeStatus(DECOMMISSIONED, HEALTHY));
     }
   }
+
+  @ParameterizedTest
+  @ValueSource(ints = {NODE_PER_RACK + 1, 2 * NODE_PER_RACK + 1})
+  public void chooseNodeWithUsedNodesMultipleRack(int datanodeCount)
+      throws SCMException {
+    assumeTrue(datanodeCount > NODE_PER_RACK);
+    setup(datanodeCount);
+    int nodeNum = 1;
+    List<DatanodeDetails> excludedNodes = new ArrayList<>();
+    List<DatanodeDetails> usedNodes = new ArrayList<>();
+
+    // 2 replicas, two existing datanodes on same rack
+    usedNodes.add(datanodes.get(0));
+    usedNodes.add(datanodes.get(1));
+
+    List<DatanodeDetails> datanodeDetails = policy.chooseDatanodes(usedNodes,
+        excludedNodes, null, nodeNum, 0, 5);
+    Assertions.assertEquals(nodeNum, datanodeDetails.size());
+
+    // New DN should be on different rack than DN0 & DN1
+    Assertions.assertTrue(!cluster.isSameParent(
+        datanodes.get(0), datanodeDetails.get(0)) &&
+        !cluster.isSameParent(datanodes.get(1), datanodeDetails.get(0)));
+
+    // 2 replicas, two existing datanodes on different rack
+    usedNodes.clear();
+    // 1st Replica on rack0
+    usedNodes.add(datanodes.get(0));
+    // 2nd Replica on rack1
+    usedNodes.add(datanodes.get(5));
+
+    datanodeDetails = policy.chooseDatanodes(usedNodes,
+        excludedNodes, null, nodeNum, 0, 5);
+    Assertions.assertEquals(nodeNum, datanodeDetails.size());
+
+    // New replica should be either on rack0 or rack1
+    Assertions.assertTrue(cluster.isSameParent(
+        datanodes.get(0), datanodeDetails.get(0)) ||
+        cluster.isSameParent(datanodes.get(1), datanodeDetails.get(0)));
```

Comment:
Should be `cluster.isSameParent(datanodes.get(5), datanodeDetails.get(0))`.

Reply:
Thanks for pointing out, updated test.